src/main/java/hudson/plugins/ec2/util/AmazonEC2Factory.java

```diff
@@ -0,0 +1,12 @@
+package hudson.plugins.ec2.util;
+
+import java.net.URL;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+
+public abstract class AmazonEC2Factory {
```

Comment:
You can make this an `interface` to simplify this class a bit.

Reply:
I had it as an interface at first but it interfered with some Guice module provided by Jenkins that was scanning all potential @Injected classes and looking at their superclasses. So when annotating an interface field with @Inject, this module would (amongst other things) run .getSuperClass() on that, which returns null for an interface, and Object for a class, resulting in a NullPointerException.

I did some refactoring after that though, so I might have built the issue away without knowing it. I'll try it out.